build_runner/lib/src/entrypoint/options.dart

```diff
@@ -98,18 +102,9 @@ class SharedOptions {
 
   factory SharedOptions.fromParsedArgs(ArgResults argResults,
       Iterable<String> positionalArgs, String rootPackage, Command command) {
-    var buildDirs = _parseBuildDirs(argResults);
+    var buildDirs = _parseBuildDirs(argResults)
+      ..addAll(_parsePositionalBuildDirs(positionalArgs, command));
```

Comment:
[optional] How about a list literal?

```dart
var buildDirs = [
  ..._parseBuildDirs(argResults),
  ..._parsePositionalBuildDirs(positionalArgs, command),
];
```

Reply:
well, in this case it would actually be less efficient and longer?

Reply:
IMO it would be more clear. In this case I didn't even get it correct, I should have used `{}` instead of `[]`. If we us a Set literal that would be explicit. We could also change both `_parse` to return an Iterable and not lose any efficiency from this version.

Reply:
now you will just wonder if its a map or set :D